kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/engine/DefaultJsonHandlerSuite.scala

```diff
@@ -436,4 +437,43 @@ class DefaultJsonHandlerSuite extends AnyFunSuite with TestUtils with DefaultVec
       writeAndVerify(overwrite = true)
     }
   }
+
+  test("parse diverse type values in a map[string, string]") {
+    val input =
+      """
+        |{
+        |   "inCommitTimestamp":1740009523401,
+        |   "timestamp":1740009523401,
+        |   "engineInfo":"myengine.com",
+        |   "operation":"WRITE",
+        |   "operationParameters":
+        |     {"mode":"Append","statsOnLoad":false,"partitionBy":"[]"},
```

Comment:
can we add a KV that is null? one case up above that I think we don't hit (value is null and we are parsing as string)

Reply:
do you know if we expect any other possible cases besides boolean? i.e. is it possible it would be `"partitionBy": []`?

Reply:
So far I see ints and boolean, the field values `operationMetrics` covers both for the tests